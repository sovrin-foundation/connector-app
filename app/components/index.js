// @flow
import React from 'react'

export { default as Badge } from './badge'
export { default as InfoSectionList } from './info-section-list'
export { ListItem, ListItemData } from './info-section-list'
export { Avatar } from './avatar/avatar'
export { BadgeAvatar } from './avatar/badge-avatar'
export {
  Container,
  CustomView,
  headerStyles,
  CustomSafeAreaView,
} from './layout'
export { default as Icon } from './icon'
export { default as CustomText, styles as TextStyles } from './text'
export { default as CustomButton } from './button'
export { default as QRScanner } from './qr-scanner/qr-scanner'
export { default as TouchId } from './touch-id/touch-id'
export { default as Request } from './request/request'
export { default as PinCodeBox } from './pin-code/pin-code-box'
export {
  default as ImageColorPicker,
} from './image-color-picker/image-color-picker'
export {
  default as ConnectionTheme,
} from './image-color-picker/connection-theme'
export { default as AvatarsPair } from './AvatarsPair'
export { default as CustomModal } from './CustomModal'
export {
  default as ClaimProofHeader,
} from './claim-proof-header/claim-proof-header'
export { default as Separator } from './separator'
export { default as FooterActions } from './footer-actions/footer-actions'
export { default as CustomList } from './custom-list/custom-list'
export { default as CustomDate } from './custom-date/custom-date'
export { default as UserAvatar } from './user-avatar/user-avatar'
export { default as Keyboard } from './keyboard/keyboard'
export { default as CustomHeader } from './custom-header/custom-header'
export { default as Loader } from './loader/loader'
