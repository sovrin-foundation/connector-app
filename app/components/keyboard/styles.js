// @flow

import { StyleSheet, Dimensions, Platform } from 'react-native'
import {
  isBiggerThanShortDevice,
  deviceHeight,
  MEDIUM_DEVICE,
} from '../../common/styles/constant'
import { PUSH_NOTIFICATION_PERMISSION } from '../../push-notification/type-push-notification'

const ROW_MARGIN = isBiggerThanShortDevice ? 36 : 15
const FIRST_ROW_MARGIN = isBiggerThanShortDevice ? 24 : 8

const ROW_HEIGHT =
  Platform.OS === 'ios' ? 36 : deviceHeight < MEDIUM_DEVICE ? 27 : 36

const FONT_SIZE = isBiggerThanShortDevice ? 28 : 26
export const INPUT_RANGE = [0, 0.5, 1]
export const OUTPUT_RANGE =
  Platform.OS === 'ios'
    ? isBiggerThanShortDevice ? [28, 29, 30] : [26, 30, 34]
    : deviceHeight < MEDIUM_DEVICE ? [28, 30, 32] : [26, 30, 34]

export const styles = StyleSheet.create({
  container: {
    marginHorizontal: 32, // TODO: update style guide to reflect new constant
    marginBottom: 5,
  },
  firstRow: {
    height: ROW_HEIGHT,
    marginTop: FIRST_ROW_MARGIN,
  },
  row: {
    height: ROW_HEIGHT,
    marginTop: ROW_MARGIN,
  },
  number: {
    fontSize: FONT_SIZE,
    fontWeight: '500', // TODO: update style guide to reflect new constant
  },
})
