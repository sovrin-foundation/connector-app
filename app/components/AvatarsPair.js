// @flow
import React, { PureComponent } from 'react'
import { Image, StyleSheet, Platform } from 'react-native'
import Modal from 'react-native-modal'
import { Avatar, CustomView, UserAvatar } from '../components'
import { color, OFFSET_1X, OFFSET_2X, isiPhone5 } from '../common/styles'
import avatarStyle from '../components/avatar/avatar-style'
import type { ImageSource } from '../common/type-common'

// TODO:KS Add types and tests for this component
// change the name of the file to match our convention of snake-case
// move this component to it's own folder, add type, tests, etc.
export default class AvatarsPair extends PureComponent<*, void> {
  renderAvatarWithSource = (avatarSource: ImageSource | number) => (
    <Avatar medium src={avatarSource} testID={`avatars-pair-avatars-invitee`} />
  )

  render() {
    const { middleImage, middleImageStyle, avatarRight, testID } = this.props
    return (
      <CustomView
        row
        vCenter
        spaceBetween
        style={[styles.avatarsContainer]}
        testID={`${testID}-text-avatars-container`}
      >
        <UserAvatar>{this.renderAvatarWithSource}</UserAvatar>
        {middleImage && (
          <Image
            style={[styles.checkMark, middleImageStyle && middleImageStyle]}
            source={middleImage}
            testID={`${testID}-check-mark`}
            accessible={true}
            accessibilityLabel={`${testID}-check-mark`}
          />
        )}
        {avatarRight && (
          <Avatar
            medium
            src={avatarRight}
            testID={`${testID}-text-avatars-inviter`}
          />
        )}
      </CustomView>
    )
  }
}

const styles = StyleSheet.create({
  avatarsContainer: {
    marginVertical: OFFSET_1X,
    marginHorizontal: isiPhone5 ? OFFSET_1X : OFFSET_2X,
    ...Platform.select({
      android: {
        paddingVertical: OFFSET_1X / 2,
      },
    }),
  },
  checkMark: {
    width: 30,
    height: 22,
  },
})
