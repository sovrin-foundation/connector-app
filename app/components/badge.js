// @flow
import React from 'react'
import { Image, Text, StyleSheet, TouchableWithoutFeedback } from 'react-native'
import { CustomView } from './layout/custom-view'
import { color, font } from '../common/styles/constant'
import empty from '../common/empty'

// We need both type of ribbon and it is better to load them once
const whiteRibbon = require('../images/ribbon_white.png')
const grayRibbon = require('../images/ribbon.png')

// TODO:KS Add props, this component is not being used right now
// so it is not that important to add types here
export const BadgeLabel = (props: *) => {
  const textColor = props.secondary ? styles.secondary : styles.primary
  const center = props.center ? styles.center : null

  return (
    <Text {...props} style={[styles.label, textColor, center, props.style]}>
      {props.text}
    </Text>
  )
}

const Badge = ({ count, secondary, badgeStyle, shadow, onPress }: *) => {
  const badgeColor = secondary
    ? color.bg.primary.font.primary
    : color.bg.secondary.font.primary
  const ribbon = secondary ? whiteRibbon : grayRibbon
  const style = shadow ? styles.shadow : null

  return (
    <CustomView clearBg style={[badgeStyle, style]}>
      <TouchableWithoutFeedback onPress={onPress}>
        <Image
          resizeMode="contain"
          source={ribbon}
          style={styles.image}
          onPress={onPress}
        />
      </TouchableWithoutFeedback>
      <BadgeLabel center secondary={secondary} text={count} style={empty} />
    </CustomView>
  )
}

const styles = StyleSheet.create({
  image: {
    width: 22,
    height: 26,
  },
  shadow: {
    shadowColor: color.bg.primary.color,
    shadowOffset: {
      width: 1,
      height: 1,
    },
    shadowRadius: 4,
    shadowOpacity: 0.9,
  },
  label: {
    fontSize: font.size.XS,
    fontWeight: 'bold',
  },
  primary: {
    color: color.bg.secondary.font.tertiary,
  },
  secondary: {
    color: color.bg.primary.font.primary,
  },
  center: {
    textAlign: 'center',
    textAlignVertical: 'center',
  },
})

export default Badge
