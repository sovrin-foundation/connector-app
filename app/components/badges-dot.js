// @flow
import React, { Component, PureComponent } from 'react'
import { View, StyleSheet, Platform } from 'react-native'
import { HAIRLINE_WIDTH, maroonRed, white } from '../common/styles/constant'

type DotProps = {
  size: string,
}

export class Dot extends PureComponent<DotProps, void> {
  render() {
    const { size } = this.props
    const dotSize =
      size === 'small'
        ? styles.badgeSmall
        : size === 'medium' ? styles.badgeMedium : styles.badgeLarge
    return <View style={[styles.badge, dotSize]} />
  }
}

const styles = StyleSheet.create({
  badge: {
    backgroundColor: maroonRed,
    ...Platform.select({
      ios: {
        borderWidth: HAIRLINE_WIDTH,
      },
      android: {
        borderWidth: 1,
      },
    }),
    borderColor: white,
  },
  badgeSmall: {
    width: 15,
    height: 15,
    borderRadius: 10,
  },
  badgeMedium: {
    width: 20,
    height: 20,
    borderRadius: 10,
  },
  badgeLarge: {
    width: 25,
    height: 25,
    borderRadius: 15,
  },
})
