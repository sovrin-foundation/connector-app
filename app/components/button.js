// @flow
import React, { PureComponent } from 'react'
import { StyleSheet } from 'react-native'
import { Button } from 'react-native-elements'
import { color, font } from '../common/styles/constant'
import debounce from 'lodash.debounce'

const empty = []

const getButtonProps = type => ({
  backgroundColor: color.actions[type],
  color: color.actions.font[type],
  textAlign: 'center',
})

// TODO:KS Add specific prop that are needed and used by Button
export default class CustomButton extends PureComponent<*, void> {
  debounceButtonPress = debounce(
    event => {
      if (this.props.onPress) {
        this.props.onPress(event)
      }
    },
    600,
    { leading: true, trailing: false }
  )

  render() {
    const {
      primary,
      secondary,
      tertiary,
      quaternary,
      fifth,
      dangerous,
      medium,
      customColor = {},
      disabled,
    } = this.props
    const buttonStyles = this.props.style || empty
    const style = [
      medium ? styles.mediumVerticalPadding : null,
      ...buttonStyles,
    ]

    const buttonType = primary
      ? 'sixth'
      : secondary
        ? 'secondary'
        : tertiary
          ? 'tertiary'
          : quaternary
            ? 'quaternary'
            : dangerous ? 'dangerous' : fifth ? 'eighth' : 'fifth'
    const buttonProps = { ...getButtonProps(buttonType), ...customColor }
    // when button is disabled, we want to apply same color that is
    // generated while picking up the color from image
    const disabledStyles = [
      { backgroundColor: customColor.backgroundColor || color.actions.none },
      styles.disabled,
    ]

    return (
      <Button
        {...this.props}
        onPress={this.debounceButtonPress}
        {...buttonProps}
        accessible={true}
        accessibilityLabel={this.props.testID}
        buttonStyle={style}
        containerViewStyle={styles.buttonContainer}
        disabledStyle={disabledStyles}
      />
    )
  }
}

const styles = StyleSheet.create({
  buttonContainer: {
    marginRight: 0,
    marginLeft: 0,
  },
  mediumVerticalPadding: {
    paddingVertical: 17,
  },
  disabled: {
    opacity: 0.4,
  },
  text: {
    fontWeight: '600',
    fontSize: font.size.M,
  },
})
