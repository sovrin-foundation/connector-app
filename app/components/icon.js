// @flow
import React from 'react'
import { StyleSheet, Image, TouchableWithoutFeedback, View } from 'react-native'
import { CustomView } from './layout/custom-view'
import type { GenericObject } from '../common/type-common'
import { white } from '../common/styles'

const empty = []

// TODO:KS Change GenericObject to actual props of icon
const Icon = (props: GenericObject) => {
  const {
    src,
    small,
    medium,
    mediumLarge,
    large,
    extraLarge,
    xxLarge,
    iconStyle = empty,
    testID,
    resizeMode = 'contain',
    halo,
    round,
    haloStyle = empty,
    onPress,
    onLongPress,
    backgroundRoundWhite,
  } = props
  const size = small
    ? 'small'
    : medium
      ? 'medium'
      : extraLarge
        ? 'extraLarge'
        : mediumLarge ? 'mediumLarge' : xxLarge ? 'xxLarge' : 'large'
  const roundImageStyle =
    halo || round
      ? // $FlowFixMe flow does not support dynamic object key
        styles[`${size}RoundIcon`]
      : null
  let filteredProps = {}
  if (typeof props.testID !== 'undefined') {
    filteredProps.testID = testID
    filteredProps.accessible = true
    filteredProps.accessibilityLabel = testID
  }
  const backgroundRoundWhiteImage = backgroundRoundWhite
    ? // $FlowFixMe flow does not support dynamic object key
      [styles.backgroundWhite, styles[`${size}RoundIcon`]]
    : {}
  const iconImage = (
    <View style={backgroundRoundWhiteImage}>
      <Image
        source={src}
        style={[styles[size], roundImageStyle, ...iconStyle]}
        resizeMode={resizeMode}
        {...filteredProps}
      />
    </View>
  )

  let haloView
  if (halo) {
    // $FlowFixMe flow does not support dynamic object lookup
    const haloSizeStyle = styles[`${size}Halo`]
    haloView = (
      <CustomView center style={[haloSizeStyle, haloStyle]}>
        {iconImage}
      </CustomView>
    )
  }

  return (
    <CustomView clearBg {...props}>
      {halo ? haloView : iconImage}
    </CustomView>
  )
}

export default Icon

export const size = {
  small: 20,
  medium: 30,
  mediumLarge: 40,
  large: 50,
  extraLarge: 80,
  xxLarge: 90,
}

const styles = StyleSheet.create({
  small: {
    width: size.small,
    height: size.small,
  },
  medium: {
    width: size.medium,
    height: size.medium,
  },
  mediumLarge: {
    width: size.mediumLarge,
    height: size.mediumLarge,
  },
  large: {
    width: size.large,
    height: size.large,
  },
  extraLarge: {
    width: size.extraLarge,
    height: size.extraLarge,
  },
  xxLarge: {
    width: size.xxLarge,
    height: size.xxLarge,
  },
  smallRoundIcon: {
    borderRadius: size.small / 2,
  },
  mediumRoundIcon: {
    borderRadius: size.medium / 2,
  },
  largeRoundIcon: {
    borderRadius: size.large / 2,
  },
  mediumLargeRoundIcon: {
    borderRadius: size.mediumLarge / 2,
  },
  extraLargeRoundIcon: {
    borderRadius: size.extraLarge / 2,
  },
  extraLargeHalo: {
    // we are assuming that halo will be at least 20pt larger than image
    borderRadius: (size.extraLarge + 20) / 2,
    width: size.extraLarge + 20,
    height: size.extraLarge + 20,
    backgroundColor: 'rgba(0,0,0,0.07)',
  },
  backgroundWhite: {
    backgroundColor: white,
  },
})
