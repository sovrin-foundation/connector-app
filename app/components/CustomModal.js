// @flow
import React, { PureComponent } from 'react'
import { View, Image, StyleSheet, Platform } from 'react-native'
import Modal from 'react-native-modal'
import {
  color,
  HAIRLINE_WIDTH,
  OFFSET_1X,
  OFFSET_3X,
  isiPhone5,
} from '../common/styles'
import { CustomView, CustomButton } from '../components'
import debounce from 'lodash.debounce'
import { noop } from '../common'

type CustomModalProps = {
  onPress?: (event: any) => void,
  children?: any,
  disabled?: boolean,
  buttonText?: string,
  testID: string,
  isVisible: boolean,
  accessibilityLabel?: string,
  backdropOpacity?: number,
  animationIn: string,
  animationOut: string,
  animationOutTiming: number,
  onModalHide?: () => void,
  fullScreen?: boolean,
}

export default class CustomModal extends PureComponent<CustomModalProps, void> {
  static defaultProps = {
    backdropOpacity: 1,
    animationIn: 'zoomIn',
    animationOut: 'zoomOut',
    animationOutTiming: 100,
  }

  onPressDebounce = debounce(
    event => {
      if (this.props.onPress) {
        this.props.onPress(event)
      }
    },
    600,
    { leading: true, trailing: false }
  )

  render() {
    const {
      disabled,
      onPress,
      buttonText,
      testID,
      isVisible,
      accessibilityLabel = 'Continue to see your new connection',
      backdropOpacity,
      animationIn,
      animationOut,
      animationOutTiming,
      fullScreen,
      onModalHide = noop,
    } = this.props
    return (
      <Modal
        backdropColor={color.bg.tertiary.color}
        backdropOpacity={backdropOpacity}
        isVisible={isVisible}
        animationIn={animationIn}
        animationOut={animationOut}
        animationOutTiming={animationOutTiming}
        onModalHide={onModalHide}
        useNativeDriver={true}
        hideModalContentWhileAnimating={true}
      >
        {fullScreen ? (
          <CustomView fifth>
            <CustomView>{this.props.children}</CustomView>
          </CustomView>
        ) : (
          <CustomView fifth shadow style={[styles.container]}>
            <CustomView style={[styles.innerContainer]}>
              {this.props.children}
            </CustomView>
            {onPress && (
              <CustomButton
                fifth
                disabled={disabled}
                disabledStyle={[styles.disabledStyle]}
                onPress={this.onPressDebounce}
                title={buttonText}
                testID={`${testID}-success-continue`}
                textStyle={{ fontWeight: 'bold' }}
              />
            )}
          </CustomView>
        )}
      </Modal>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    marginHorizontal: isiPhone5 ? OFFSET_1X : OFFSET_3X,
  },
  innerContainer: {
    ...Platform.select({
      ios: {
        borderBottomColor: color.bg.fifth.font.tertiary,
        borderBottomWidth: HAIRLINE_WIDTH,
      },
      android: {
        borderBottomColor: color.bg.fifth.font.secondary,
        borderBottomWidth: 1,
      },
    }),
    paddingVertical: OFFSET_1X,
  },
  disabledStyle: {
    backgroundColor: 'transparent',
  },
})
