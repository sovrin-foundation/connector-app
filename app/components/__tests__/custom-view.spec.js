// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { CustomView } from '../../components'
import { PADDING_VERTICAL } from './../../common/styles/constant'

describe('<CustomView />', () => {
  it('doubleVerticalSpace should overwrite verticalSpace style if both were provided as props to customView', () => {
    // icon should have only icon props
    // and rest of the props should be passed to CustomView
    const wrapper = renderer
      .create(<CustomView verticalSpace doubleVerticalSpace />)
      .toJSON()

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.props.style.paddingVertical).toBe(PADDING_VERTICAL * 2)
  })
})
