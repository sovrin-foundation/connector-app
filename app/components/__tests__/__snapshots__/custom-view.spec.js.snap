// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomView /> doubleVerticalSpace should overwrite verticalSpace style if both were provided as props to customView 1`] = `
<View
  doubleVerticalSpace={true}
  style={
    Object {
      "paddingVertical": 16,
    }
  }
  verticalSpace={true}
/>
`;
